types/firestore.d.ts

```diff
@@ -2424,6 +2437,127 @@ declare namespace FirebaseFirestore {
+     *
+     * @param {...Filter} filters  Optional. The {@link Filter}s
+     * for OR operation. These must be created with calls to {@link Filter#where},
+     * {@link Filter#or}, or {@link Filter#and}.
+     * @returns {Filter} The created {@link Filter}.
+     *
+     * @example
+     * ```
+     * let collectionRef = firestore.collection('col');
+     *
+     * // doc.foo == 'bar' || doc.baz > 0
+     * let orFilter = Filter.or(Filter.where('foo', '==', 'bar'), Filter.where('baz', '>', 0));
+     *
+     * collectionRef.where(orFilter).get().then(querySnapshot => {
+     *   querySnapshot.forEach(documentSnapshot => {
+     *     console.log(`Found document at ${documentSnapshot.ref.path}`);
+     *   });
+     * });
+     * ```
+     */
+    static or(...filters: Filter[]): Filter;
+
+    /**
+     * Creates and returns a new [CompositeFilter]{@link CompositeFilter} that performs
+     * a logical AND of all the provided {@link Filter}s. The returned Filter can be
+     * applied to [Query.where()]{@link Query#where}, [Filter.or()]{@link Filter#or},
+     * or [Filter.and()]{@link Filter#and}. When applied to a [Query]{@link Query}
+     * it requires that documents must satisfy all of the provided {@link Filter}s.
+     *
+     * @param {...Filter} filters  Optional. The {@link Filter}s
+     * for OR operation. These must be created with calls to {@link Filter#where},
+     * {@link Filter#or}, or {@link Filter#and}.
+     * @returns {Filter} The created {@link Filter}.
+     *
+     * @example
+     * ```
+     * let collectionRef = firestore.collection('col');
+     *
+     * // doc.foo == 'bar' && doc.baz > 0
+     * let orFilter = Filter.and(Filter.where('foo', '==', 'bar'), Filter.where('baz', '>', 0));
+     *
+     * collectionRef.where(orFilter).get().then(querySnapshot => {
+     *   querySnapshot.forEach(documentSnapshot => {
+     *     console.log(`Found document at ${documentSnapshot.ref.path}`);
+     *   });
+     * });
+     * ```
+     */
+    static and(...filters: Filter[]): Filter;
+  }
+
+  /**
+   * A `UnaryFilter` is used to narrow the set of documents returned by
+   * a Firestore query by filtering on one or more document fields.
+   * `UnaryFilter`s are created by invoking {@link Filter#where} and can then
+   * be passed to {@link Query#where} to create a new {@link Query} instance
+   * that also contains this `UnaryFilter`.
+   */
+  export class UnaryFilter extends Filter {
+    private constructor();
+  }
+  
+  /**
+   * A `CompositeFilter` is used to narrow the set of documents returned
+   * by a Firestore query by performing the logical OR or AND of multiple
+   * {@link Filters}s. `CompositeFilters`s are created by invoking {@link Filter#or}
+   * or {@link Filter#and} and can then be passed to {@link Query#where}
+   * to create a new query instance that also contains the `CompositeFilter`.
+   */
+  export class CompositeFilter extends Filter {
+    private constructor(f);
+  }
```

Comment:
Do we need to expose UnaryFilter and CompositeFilter?

Reply:
Removed this from d.ts and API reference docs.